tests/gnrc_tx_sync/main.c

```diff
@@ -0,0 +1,220 @@
+{
+    (void)dev;
+    const uint8_t addr[] = {0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37};
+    if (!IS_USED(MODULE_NETDEV_IEEE802154)) {
+        return -ENOTSUP;
+    }
+    assert(max_len >= sizeof(addr));
+    memcpy(value, addr, sizeof(addr));
+    return sizeof(addr);
+}
+
+static int netdev_get_src_len(netdev_t *dev, void *value, size_t max_len)
+{
+    (void)dev;
+    const uint16_t src_len = 2;
+    if (!IS_USED(MODULE_NETDEV_IEEE802154)) {
+        return -ENOTSUP;
+    }
+    assert(max_len == sizeof(src_len));
+    memcpy(value, &src_len, sizeof(src_len));
+    return sizeof(src_len);
+}
+
+int main(void)
+{
+    puts(
+        "Test application for gnrc_tx_sync\n"
+        "=================================\n"
+        "\n"
+        "This application sends a single UDP datagram over a virtual Ethernet\n"
+        "or 802.15.4 device. (Set TEST_6LO to 1 for 802.15.4 and to 0 for\n"
+        "Ethernet.) The simulated network devices blocks for 100ms - more than\n"
+        "enough time for sock_upd_send() to return if no synchronization is done.\n"
+        "If an 802.15.4 device is simulated, the test datagram will be transmitted\n"
+        "in two fragments. In that case sock_udp_send() must only return after both\n"
+        "fragments are out. In the Ethernet case, only a single frame is send.\n"
+    );
+
+    if (IS_USED(MODULE_NETDEV_IEEE802154)) {
+        puts("IEEE 802.15.4 mode (TEST_6LO=1), sending 2 6LoWPAN fragments");
+    }
+    else {
+        puts("Ethernet mode (TEST_6LO=0), sending 1 IPv6 packet");
+    }
+    msg_init_queue(main_msg_queue, MAIN_QUEUE_SIZE);
+    netdev_test_setup(&netdev_test, NULL);
+    netdev_test_set_send_cb(&netdev_test, netdev_send);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_DEVICE_TYPE, netdev_get_device_type);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_MAX_PDU_SIZE, netdev_get_max_pdu_size);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_PROTO, netdev_get_proto);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_ADDRESS, netdev_get_address);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_ADDRESS_LONG, netdev_get_address_long);
+    netdev_test_set_get_cb(&netdev_test, NETOPT_SRC_LEN, netdev_get_src_len);
+    gnrc_netif_raw_create(&netif, netif_stack, sizeof(netif_stack), NETIF_PRIO,
+                          "netdev_test", netdev);
+
+    sock_udp_t sock;
+    sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+    sock_udp_ep_t remote = { .family = AF_INET6 };
+    remote.port = 12345;
+    ipv6_addr_set_all_nodes_multicast((ipv6_addr_t *)&remote.addr.ipv6,
+                                      IPV6_ADDR_MCAST_SCP_LINK_LOCAL);
+    expect(sock_udp_create(&sock, &local, NULL, 0) == 0);
+    /* With 6LoWPAN, This test message needs exactly two 32 byte fragments to be transmitted */
+    static const char test_msg[33] = { 'T', 'e', 's', 't' };
+    expect(sock_udp_send(&sock, test_msg, sizeof(test_msg), &remote) > 0);
```

Comment:
Maybe add a comment, that _this_ is actually the test here (that `send` only returns _after_ `sends_complete` is the expected value), otherwise, this test could be understand as a run-of-the-mill `sock_udp_send` test ;-). I know, this is basically what is described in the boilerplate text, but IMHO it does not become immediately clear which part of the code does what.

Reply:
Done